### PR TITLE
Implement postsubmit workflow for gke-integration

### PR DESCRIPTION
**Pull request by Demonsthere**
**Description**
Cleanup previous task and consolidate scripts into a single one to operate on PR/commit depending on labels. 

**Related issue(s)**
#150

### CODE REVIEW

File: prow/scripts/cluster-integration/kyma-gke-integration.sh [hunk 1]
```diff
@@ -7,7 +7,7 @@
 #
 # - REPO_OWNER - Set up by prow, repository owner/organization
 # - REPO_NAME - Set up by prow, repository name
-# - PULL_NUMBER - Set up by prow, Pull request number
+# - BUILD_TYPE - Set up by prow, pr/master/release
```
Comment (suleymanakbas91) [inline prow/scripts/cluster-integration/kyma-gke-integration.sh]: `BUILD_TYPE` is not set by prow

Comment (Demonsthere) [reply prow/scripts/cluster-integration/kyma-gke-integration.sh]: Technically speaking it is set by the preset-labels which are managed by prow. We have `preset-build-master: "true"` and `preset-build-pr: "true"` which set `BUILD_TYPE` to either `pr/master/release`

File: prow/jobs/kyma/kyma-integration.yaml [hunk 2]
```diff
@@ -74,6 +74,34 @@ postsubmits:
           - image: eu.gcr.io/kyma-project/prow/bootstrap:0.0.1
             command:
               - "/home/prow/go/src/github.com/kyma-project/test-infra/prow/scripts/provision-vm-and-start-kyma.sh"
+    - name: kyma-gke-integration-master
```
Comment (aszecowka) [inline prow/jobs/kyma/kyma-integration.yaml]: in the Migration guide, we have the same name for presubmit and postsubmit job. Is there any reason to keep sufix master?

Comment (aszecowka) [inline prow/jobs/kyma/kyma-integration.yaml]: wny don't you extract common part to job_template (see migration guide)

File: prow/scripts/cluster-integration/kyma-gke-integration.sh (hunk 1 above)
Comment (aszecowka) [reply prow/scripts/cluster-integration/kyma-gke-integration.sh]: IMO we should mention that it is set by specyfing given preset in job definition.

File: prow/scripts/cluster-integration/kyma-gke-integration.sh [hunk 3]
```diff
@@ -131,7 +149,7 @@ init
 
 shout "Build Kyma-Installer Docker image"
 date
-export KYMA_INSTALLER_IMAGE="${DOCKER_PUSH_REPOSITORY}${DOCKER_PUSH_DIRECTORY}/gke-integration/${REPO_OWNER}/${REPO_NAME}:PR-${PULL_NUMBER}"
+export KYMA_INSTALLER_IMAGE
```
Comment (aszecowka) [inline prow/scripts/cluster-integration/kyma-gke-integration.sh]: move this up, where this var is set

File: prow/scripts/cluster-integration/kyma-gke-integration.sh [hunk 4]
```diff
@@ -30,7 +30,7 @@ set -o errexit
 
 discoverUnsetVar=false
 
-for var in REPO_OWNER REPO_NAME PULL_NUMBER DOCKER_PUSH_REPOSITORY DOCKER_PUSH_DIRECTORY KYMA_PROJECT_DIR CLOUDSDK_CORE_PROJECT CLOUDSDK_COMPUTE_REGION CLOUDSDK_DNS_ZONE_NAME GOOGLE_APPLICATION_CREDENTIALS; do
```
Comment (aszecowka) [inline prow/scripts/cluster-integration/kyma-gke-integration.sh]: update documentation that PULL_NUMBER is not required

File: prow/scripts/cluster-integration/kyma-gke-integration.sh [hunk 5]
```diff
@@ -97,13 +97,31 @@ cleanup() {
...
 
+# Enforce lowercase
+readonly REPO_OWNER=$(echo "${REPO_OWNER}" | tr '[:upper:]' '[:lower:]')
+export REPO_OWNER
+readonly REPO_NAME=$(echo "${REPO_NAME}" | tr '[:upper:]' '[:lower:]')
+export REPO_NAME
+
+if [[ "$BUILD_TYPE" == "pr" ]]; then
+    # In case of PR, operate on PR number
+    IP_ADDRESS_NAME=$(echo "pr-${PULL_NUMBER}-job-${PROW_JOB_ID}" | tr "[:upper:]" "[:lower:]")
```
Comment (aszecowka) [inline prow/scripts/cluster-integration/kyma-gke-integration.sh]: may be here we should check if PULL_NUMBER is set

File: prow/scripts/cluster-integration/kyma-gke-integration.sh [hunk 6]
```diff
@@ -97,13 +97,31 @@ cleanup() {
...
+    CLUSTER_NAME="gkeint-${REPO_OWNER}-${REPO_NAME}-${PULL_NUMBER}"
+    KYMA_INSTALLER_IMAGE="${DOCKER_PUSH_REPOSITORY}${DOCKER_PUSH_DIRECTORY}/gke-integration/${REPO_OWNER}/${REPO_NAME}:PR-${PULL_NUMBER}"
+else 
+    # Otherwise (master, or release), operate on triggering commit id
+    readonly COMMIT_ID=$(cd "$KYMA_SOURCES_DIR" && git rev-parse --short HEAD)
```
Comment (aszecowka) [inline prow/scripts/cluster-integration/kyma-gke-integration.sh]: could you check if we can replace COMMIT_ID with PULL_BASE_SHA provided by Prow?

File: prow/jobs/kyma/kyma-integration.yaml [hunk 7]
```diff
@@ -51,7 +51,7 @@ presubmits: # runs on PRs
               - "bash"
             args:
               - "-c"
-              - "${KYMA_PROJECT_DIR}/test-infra/prow/scripts/cluster-integration/pr-gke-integration.sh"
```
Comment (aszecowka) [inline prow/jobs/kyma/kyma-integration.yaml]: README is out-dated.

File: prow/jobs/kyma/kyma-integration.yaml (hunk 2 above)
Comment (Demonsthere) [reply prow/jobs/kyma/kyma-integration.yaml]: The existing presubmit jobs are `kyma-integration` and `kyma-gke-integration`, and the existing postsubmit job was `kyma-integration-master`. I created `kyma-gke-integration-master` as it seemed consistent with the existing naming

File: prow/scripts/cluster-integration/kyma-gke-integration.sh (hunk 6 above)
Comment (Demonsthere) [reply prow/scripts/cluster-integration/kyma-gke-integration.sh]: We could, however the env looks like `PULL_BASE_SHA=b25b58b147fab0c89413ca238858e17b7916e201`, so we would have to trim it

File: prow/jobs/kyma/kyma-integration.yaml (hunk 2 above)
Comment (aszecowka) [reply prow/jobs/kyma/kyma-integration.yaml]: I understand that, but maybe in your PR we can make it consistent with other jobs?

**Comment by Demonsthere** [comment]
Closing due to a rebase error.

**Pull request closed without merging**